docker-compose.dev.local.override.yml

```diff
@@ -4,8 +4,11 @@
 
 services:
   web:
-    command: [ "bash", "-c",
-               "git config --global --add safe.directory /app && pip3 install -U -e . && flask init-db && flask run --host=0.0.0.0 --debug" ]
+    command: [
+      "/sbin/my_init", "--",
```

Comment:
why just not to retain the same ENTRYPOINT of `docker.io/phusion/baseimage:jammy-1.0.1` which is the `/sbin/my_init` AFAIK and then just have here the `"bash", "-c", ...`?

Reply:
but otherwise -- looks good if that works ;)  please make it sweat (run the population script again) and see if pids are staying under control with no zombies

Reply:
> why just not to retain the same ENTRYPOINT of `docker.io/phusion/baseimage:jammy-1.0.1` which is the `/sbin/my_init` AFAIK and then just have here the `"bash", "-c", ...`?

The `docker.io/phusion/baseimage:jammy-1.0.1`  image actually doesn't have an ENTRYPOINT of `"/sbin/my_init"`. It only have `CMD ["/sbin/my_init"]` as the default command to execute, which is to be overridden.

I have considered adding `ENTRYPOINT ["/sbin/my_init"]` to our project image, but that would require starting each command to be executed with `"--"`. For example, the command you referenced above would have to be `command: [
      "--",
      "bash", "-c",
      "git config --global --add safe.directory /app && pip3 install -U -e . && flask init-db && exec flask run --host=0.0.0.0 --debug"
    ]`. I think that makes the code less clear for it is not immediately clear what the `"--"` is for, and one has to remember to use `"--"` before specifying any command to be executed because of a hidden context (the entry point) in the project Docker image.

On the other hand, we can add ``ENTRYPOINT ["/sbin/my_init", "--"]` to our project image and specify commands to be executed the way you suggested. However, doing so makes passing options to `"/sbin/my_init"` much less convenient (one has to override the entry point to pass any options to `"/sbin/my_init"` itself).

Because of the above considerations, I think we should keep the current arrangement. It is a bit longer, but it is explicit and convenient to pass options to `"/sbin/my_init"` itself.

Reply:
> please make it sweat (run the population script again) and see if pids are staying under control with no zombies

I want to sweat the server by deploying a solution for https://github.com/datalad/datalad-registry/issues/271 instead. I think we can see zombies if this PR doesn't work when adding all the gin datasets to the registry (the worker process has already accumulated some zombies over the vacation, about 40, just by updating the registered datasets).

Reply:
ok